console/console-init/ui/src/modules/iot-device/components/AddDeviceWithJson/AddDeviceWithJson.tsx

```diff
@@ -43,51 +48,110 @@ const styles = StyleSheet.create({
     paddingLeft: 30
   }
 });
+
+export interface IRegistration {
+  enabled?: boolean;
+  via?: string[];
+  ext?: any;
+  viaGroups?: string[];
+  memberOf?: string[];
+  defaults?: any;
+}
 interface IAddDeviceWithJsonProps {
-  deviceDetail?: string;
-  setDeviceDetail: (detail?: string) => void;
+  deviceDetail?: IDeviceResponse;
   onLeave: () => void;
   onSave: (detail: string) => void;
   allowTemplate?: boolean;
 }
+const getDeviceRegistrationString = (deviceDetail: IDeviceResponse) => {
+  let device: any = { registration: {} };
+  if (deviceDetail.registration) {
+    const {
+      enabled,
+      defaults,
+      memberOf,
+      via,
+      ext,
+      viaGroups
+    } = deviceDetail.registration;
+    const deviceRegistration: IRegistration = {};
+    if (enabled) {
+      deviceRegistration.enabled = enabled;
+    }
+    if (defaults) {
+      deviceRegistration.defaults = JSON.parse(defaults);
+    }
+    if (ext) {
+      deviceRegistration.ext = JSON.parse(ext);
+    }
+    if ((!via || via.length === 0) && (!viaGroups || viaGroups?.length === 0)) {
+      device.credentials = [];
+    } else {
+      if (via) {
+        deviceRegistration.via = via;
+      }
+      if (viaGroups) {
+        deviceRegistration.viaGroups = viaGroups;
+      }
+    }
+    if (memberOf && memberOf.length > 0) {
+      deviceRegistration.memberOf = memberOf;
+    }
+    if ((!via || via.length === 0) && (!viaGroups || viaGroups?.length === 0)) {
+      device.credentials = [];
+    }
+    device.registration = deviceRegistration;
+  }
+  return JSON.stringify(device, undefined, 2);
```

Comment:
It would be better to stringify where it required not here so that this function will be generic not specific.
Just return device.

Reply:
I don't think the function is for generic purpose. It is a specific function. If it found to be a generic function then we can update this function